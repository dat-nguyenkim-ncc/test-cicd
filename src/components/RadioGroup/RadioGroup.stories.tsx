import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'

import RadioGroup, { RadioGroupProps } from './RadioGroup'
import { Section } from '../primitives'

export default {
  title: 'Form Elements/RadioGroup',
  component: RadioGroup,
} as Meta

const Template: Story<RadioGroupProps> = args => (
  <Section>
    <RadioGroup {...args} />
  </Section>
)

export const Default = Template.bind({})
Default.args = {
  label: 'FinTech Type',
  name: 'fintech_type',
  options: [
    { label: 'Enabler', value: 'enabled' },
    { label: 'Disruptor', value: 'disruptor' },
  ],
} as RadioGroupProps
