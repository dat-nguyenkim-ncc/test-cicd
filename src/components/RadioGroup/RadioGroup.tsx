import React, { useState } from 'react'
import { Box, Flex, Label } from 'theme-ui'
import { Radio } from '..'
import { ViewInterface } from '../../types'

type onChangeEvent = {
  name: string
  value: string
}

export type RadioGroupProps = ViewInterface<{
  label?: string
  name: string
  options: {
    label: string
    value: string
  }[]
  onChange?(e: onChangeEvent | undefined): void
  defaultValue?: string
}>

const RadioGroup = ({ name, sx, label, options, onChange, defaultValue }: RadioGroupProps) => {
  const [selected, setSelected] = useState<string | undefined>(defaultValue)

  const onClickRadio = (value: string) => {
    setSelected(selected === value ? undefined : value)
    onChange && onChange({ name, value })
  }
  return (
    <Box sx={{ flex: 1, ...sx }}>
      {label && <Label>{label}</Label>}
      <Flex>
        {options.map((o, index) => (
          <Radio
            selected={selected === o.value}
            onClick={onClickRadio}
            sx={{ ml: index > 0 ? 6 : 0 }}
            key={index}
            {...o}
          />
        ))}
      </Flex>
    </Box>
  )
}

export default RadioGroup
