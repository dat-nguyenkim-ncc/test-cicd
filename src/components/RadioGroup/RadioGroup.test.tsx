import * as React from 'react'
import { shallow } from 'enzyme'
import RadioGroup from './RadioGroup'

describe('RadioGroup', () => {
  test('matches snapshot', () => {
    const wrapper = shallow(
      <RadioGroup
        label="Radio Group"
        name="radio"
        options={[
          { label: 'Enabler', value: 'enabled' },
          { label: 'Disruptor', value: 'disruptor' },
        ]}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
