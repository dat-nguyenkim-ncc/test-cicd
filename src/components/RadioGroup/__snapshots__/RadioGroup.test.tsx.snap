// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroup matches snapshot 1`] = `ShallowWrapper {}`;
